app/Http/Controllers/StatusController.php

```diff
@@ -142,8 +145,13 @@ public static function getGlobalDashboard(): Paginator {
                      ->join('users', 'statuses.user_id', '=', 'users.id')
                      ->where(function($query) {
                          $query->where('users.private_profile', 0)
+                               ->where('visibility', 0)
                                ->orWhere('users.id', auth()->user()->id)
-                               ->orWhereIn('users.id', auth()->user()->follows()->select('follow_id'));
+                               ->orWhere(function($query) {
+                                   $query->where('visibility', 2)
+                                         ->whereIn('users.id', auth()->user()->follows()->select('follow_id'))
+                                         ->orWhere('visibility', 0);
```

Comment:
```suggestion
                                   $query->whereIn('visibility', [0, 2])
                                         ->whereIn('users.id', auth()->user()->follows()->select('follow_id'));
```

Reply:
this will not work, since it's checking `(visibiility == 2 && users.id in follows) || visibility == 0`

Reply:
That's why I like brackets. Then you understand it better. 😐